idea/src/org/jetbrains/kotlin/idea/intentions/MergeElseIfIntention.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2010-2019 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.idea.intentions
+
+import com.intellij.openapi.editor.Editor
+import org.jetbrains.kotlin.psi.*
+
+
+class MergeElseIfIntention : SelfTargetingIntention<KtIfExpression>(KtIfExpression::class.java, "Merge 'else if'") {
+    override fun isApplicableTo(element: KtIfExpression, caretOffset: Int): Boolean {
+        val elseBody = element.`else` ?: return false
+        val nestedIf = elseBody.nestedIf() ?: return false
+        if (nestedIf.`else` != null) {
+            return false
+        }
+        return true
+    }
+
+    override fun applyTo(element: KtIfExpression, editor: Editor?) {
+        applyTo(element)
+    }
+
+    companion object {
+        fun applyTo(element: KtIfExpression): Int {
```

Comment:
Can function be inlined? Also what is the purpose of return value?

Reply:
Return value was unnecessary. Fixed it.